### PR TITLE
Fix API calls from updates

### PR DESCRIPTION
**Pull request by atabbott**
As stated in issue #90. Spotify updated web API requirements in regards to authentication. These changes were announced [here](https://developer.spotify.com/news-stories/2017/01/27/removing-unauthenticated-calls-to-the-web-api/) on 17th Jan 2017, effective 29th May 2017 it removed the ability for to make any unauthenticated calls to the API.

Currently searching appears to make unauthenticated calls to the API. A 401 error is returned with the response 'No token provided'.

Issue appears to stems from use of an old version of @jwilsson 'spotify-web-api-php' wrapper that did not include authentication token with all types of API request.

Updated 'spotify-web-api-php' to a more recent version. 

Also due to API requirement changes artist and album detail would return 401 'fetch errors' as requests were made using URLs directly without tokens. Adjusted detail fetching of album and artists to make use of methods in 'spotify-web-api-php'.

Don't fully understand the reasoning behind use of direct URLs or if certain functions relied on this implementation, but all functions still seem to work correctly as far as I can tell. 

First time looking at the codebase was approx. 2 hours ago so don't fully understand it. Therefore thought I should avoid doing too much tidying up.

### CODE REVIEW

**Comment by citelao** [comment]
@atabbott So this is a pretty big deal (as you mentioned), since the update means Spotifious now *requires* users to have an app (instead of allowing an opt out).

This is a larger change. I'm going to have to pull out all of the code that lets you opt out 😁

File: src/citelao/Spotifious/Menus/DetailAlbum.php [hunk 1]
```diff
@@ -25,13 +25,8 @@ public function __construct($options, $alfred, $api) {
...
 
-		if($api) {
-			$artistFetcher = new JsonFetcher("https://api.spotify.com/v1/albums/{$options['id']}");
-			$json = $artistFetcher->run();
-		} else {
-			$json = $api->getAlbum($options['id']);
-		}
-		
+		$json = $api->getAlbum($options['id']);
+
```
Comment (citelao) [inline src/citelao/Spotifious/Menus/DetailAlbum.php]: This was actually a typo, and is why people were affected. You should use the API if you have it, not the other way round 😮

**Comment by atabbott** [comment]
Yea, my primary goal was just writing something that would fix it for me and share it incase it was useful. Thought I shouldn't do too much messing around with the general architecture / functionality as not understanding everything I could affect something unintentionally. Plus it's your project 🙂

I didn't realise Spotifious didn't require an app previously.

**Comment by citelao** [comment]
@atabbott There's a tiny opt-out link when you setup 😆 

I've taken your changes so that any time we query the API, we do it through @jwilsson if possible. However, updating that library breaks a couple other bits of my application, so I'm going to have to bring this into my current dev branch which fixes those changes.

**Comment by atabbott** [comment]
Definitely missed the opt-out. 

Sorry about that, seemed like a pretty good one-for-one replacement from what I could see, guess resulting errors are expected though. It's not the latest version of the @jwilsson btw. Seemed to be more compatibility errors with the latest build, don't know if you want to implement that while you're fixing the other issues updating caused. 

Playlist searches sound great!

**Comment by citelao** [comment]
I might have been at a bizarre intermediate version due to how I was handling `vendor/` files. The API for fetching keys/refreshing access tokens changed at some point; I'd already done the legwork in my branch.

Thank you for contributing!

**Pull request merged**